proposals/0000-splitting-out-stable-interfaces-from-th.rst

```diff
@@ -0,0 +1,192 @@
+This can help avoid the sorts of dependency bounds propagation problems identified in the `GHC.X.Hackage proposal <https://github.com/bgamari/tech-proposals/blob/ghc-x-hackage/proposals/001-ghc-x-hackage.md>`_.
+
+There is a more subtle benefit for the ``template-haskell`` package. Currently the wide usage of ``Lift`` instances greatly limits the possible dependencies of ``template-haskell``.
+For instance, ``template-haskell`` cannot depend on ``containers`` or ``filepath``, since these libraries depend on ``template-haskell``.
+But if these packages switch to depending on our new packages, then ``template-haskell`` could depend on them.
+Currently ``template-haskell`` must vendor a small portion of ``filepath`` and ``containers``, and that would no longer be necessary.
+
+Many boot packages depend on ``template-haskell``, but all of them only depend on it for the parts of the interface exposed by ``template-haskell-lift`` and ``template-haskell-quasiquote``.
+If we can convince their maintainers to depend on these packages instead, then GHC would no longer (transitively) depend on ``template-haskell``.
+This makes it possible for packages to depend on the ``ghc`` library at the same time as a version of ``template-haskell`` different to the one bundled with that GHC.
+
+
+Proposed Change Specification
+-----------------------------
+No changes to the language or the compiler are required for this proposal.
+
+Proposed Library Change Specification
+-------------------------------------
+
+We propose to publish two new libraries: ``template-haskell-lift`` and ``template-haskell-quasiquote``.
+These will be shipped with GHC.
+They will also be buildable from Hackage.
+They will be buildable with at a *minimum* the last 3 versions of GHC.
+
+Their interfaces will be as follows:
+
+``template-haskell-lift``::
+
+   module TemplateHaskell.Lift
+    ( Q
+    , Code
+    , Quote
+    , Exp
+    , Lift(..)
+    , defaultLiftTyped -- a utility for writing `liftTyped` methods when an instance currently defines lift only
+    , liftAddrCompat -- a utility for creating an `Addr#` value, eg, for defining `Lift ByteString`
+    , liftIntCompat -- a utility for lifting an `Int` without causing issues when used with `OverloadedSyntax`
+    )
+
+``template-haskell-quasiquote``::
+
+   module TemplateHaskell.Quasiquoter
+    ( Q
+    , Exp
+    , Pat
+    , Type
+    , Dec
+    , QuasiQuoter (QuasiQuoter, quoteExp, quotePat, quoteType, quoteDec)
+    )
+
+Note that these modules are in the ``TemplateHaskell.`` namespace rather than the ``Language.Haskell.TH.`` namespace.
+The idea to use this less verbose namespace for the new stable interfaces is thanks to Adam Gundry.
+
+Effect and Interactions
+-----------------------
+This works towards removing the special case for ``template-haskell`` in (GR1), but on its own it doesn't achieve it.
```

Comment:
What is (GR1)? Do you mean (A)?

Reply:
I've clarified this. It's referring to: https://github.com/ghc-proposals/ghc-proposals/blob/01c0d736919a25e2669296aa71dd58368b09ba72/principles.rst?plain=1#L359